common/src/main/java/org/jboss/hal/testsuite/util/WizardValidator.java

```diff
@@ -11,20 +11,36 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class WizardValidator<T extends WizardWindow> {
 
-    private Set<WizardInput<T>> inputs;
-    private ConfigFragment windowFragment;
+/***
+ * Class representing a validation abstraction over wizard windows
+ * @param <T> Wizard window type
+ */
+public abstract class WizardValidator<T extends WizardWindow> {
 
-    public WizardValidator(Collection<WizardInput<T>> inputs, ConfigFragment fragment) {
-        this.inputs = new HashSet<>(inputs);
+    protected Set<WizardInput<T>> mandatoryInputs;
+    protected ConfigFragment windowFragment;
+
+    public WizardValidator(Collection<WizardInput<T>> mandatoryInputs, ConfigFragment fragment) {
+        this.mandatoryInputs = new HashSet<>(mandatoryInputs);
         this.windowFragment = fragment;
     }
 
-    public void testInputCombinationAndAssert(Collection<WizardInput<T>> presentInputs, T wizard) {
+    /**
+     * Tests all possible invalid combinations for required fields in wizard
+     * @param wizard to be tested
+     */
+    public abstract void testInvalidCombinationsAndAssert(T wizard);
```

Comment:
This is a bit more laborious than my original idea but it will work as well. We could add default implementation of this method in the future if needed.